lib/src/domain/communication/exception_extensions.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:allthenews/generated/l10n.dart';
+import 'package:flutter/material.dart';
+
+import 'api_exception.dart';
+
+extension ExceptionExtension on Exception {
+  String toErrorMessage(BuildContext context) {
+    if (this is ApiException) {
+      return _getCommunicationExceptionMessage(context);
+    } else {
+      return Strings.of(context).apiDefaultException;
+    }
+  }
+
+  String _getCommunicationExceptionMessage(context) {
```

Comment:
wiem, że private i gołym okiem wiadomo, ale mimo wszystko trzymałbym się tych jawnych typów w metodach

Reply:
to jest bardziej kwestia preferencji, mi to rybka, więc jak Ci bardziej pasują jawne typy to jak najbardziej zwracaj mi uwagę :p